quarkus-opentelemetry-exporter-gcp/integration-tests/src/main/java/io/quarkiverse/opentelemetry/exporter/it/graal/Substitutions.java

```diff
@@ -0,0 +1,153 @@
+package io.quarkiverse.opentelemetry.exporter.it.graal;
+
+import java.io.IOException;
+import java.security.GeneralSecurityException;
+import java.util.concurrent.Executor;
+import java.util.concurrent.TimeUnit;
+
+import org.threeten.bp.Duration;
+
+import com.google.api.core.ApiFunction;
+import com.google.api.gax.grpc.ChannelPrimer;
+import com.google.api.gax.grpc.GrpcHeaderInterceptor;
+import com.google.api.gax.grpc.GrpcInterceptorProvider;
+import com.google.api.gax.grpc.InstantiatingGrpcChannelProvider;
+import com.google.api.gax.rpc.HeaderProvider;
+import com.oracle.svm.core.annotate.Alias;
+import com.oracle.svm.core.annotate.Substitute;
+import com.oracle.svm.core.annotate.TargetClass;
+
+import io.grpc.CallOptions;
+import io.grpc.Channel;
+import io.grpc.ChannelCredentials;
+import io.grpc.ClientCall;
+import io.grpc.ClientInterceptor;
+import io.grpc.Grpc;
+import io.grpc.ManagedChannel;
+import io.grpc.ManagedChannelBuilder;
+import io.grpc.MethodDescriptor;
+
+/**
+ * Cut out unsupported and optional features that are only present in grpc-alts.
+ *
+ * code copied from Camel quarkus Google PubSub substitutions
+ * https://github.com/apache/camel-quarkus/commit/372e221d97e9680cebfecd8788cd3a2802eb3686
+ */
+
+final class Substitutions {
+}
+
+@TargetClass(InstantiatingGrpcChannelProvider.class)
+final class InstantiatingGrpcChannelProviderSubstitutions {
```

Comment:
These substitutions should be in the runtime module. This way, all extension usages will benefit from this, not just the integration test. 
See example: https://github.com/quarkusio/quarkus/tree/main/extensions/opentelemetry/runtime/src/main/java/io/quarkus/opentelemetry/runtime/graal